File: Python/Easy/26. Remove Duplicates from Sorted Array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i, k = 0, 1
        for j in range(1, len(nums)):
            if nums[i] < nums[j]:
                nums[i + 1] = nums[j]
                i += 1
                k += 1
        return k

File: Rust/Easy/26. Remove Duplicates from Sorted Array.rs
pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
    // Initial Solution
    if nums.len() == 1 {return 1}
    let mut k = 1;
    let mut i = 1;
    for j in 1..nums.len() {
        if j == nums.len() {return k}
        if nums[j] > nums[i - 1] {
            nums[i] = nums[j];
            i += 1;
            k += 1;
        }
    }
    k
}